wgpu-core/src/device.rs

```diff
@@ -1182,6 +1182,34 @@ impl<F: IdentityFilter<BindGroupLayoutId>> Global<F> {
         let hub = B::hub(self);
         let bindings = unsafe { slice::from_raw_parts(desc.bindings, desc.bindings_length) };
 
+        {
+            // TODO: This returns a false positive if any provided
+            // bind group layout bindings are exactly the same
+            // (including the binding number) and the length of the
+            // provided bindings equals the number of bindings we have
+            // stored. We could avoid it by allocating the provided
+            // bindings in a hashmap earlier than we do, but that
+            // allocation may end up being thrown away, if the
+            // comparison finds a compatible bind group layout here.
+            let (bind_group_layout_guard, _) = hub.bind_group_layouts.read(&mut token);
+            let bind_group_layout =
+                bind_group_layout_guard
+                    .iter(device_id.backend())
+                    .find(|(_, bgl)| {
+                        bindings.len() == bgl.bindings.len()
```

Comment:
ok, I see what you did here. Let's move the hashmap creation early and just compare hashmaps here directly. The code would be simpler, and bind group layout creation is not a place that needs optimization - throwing away the hashmap is fine.